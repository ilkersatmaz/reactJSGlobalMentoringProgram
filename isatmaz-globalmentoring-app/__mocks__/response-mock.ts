export const movieResponse = {
	data: {
		id: 221122,
		title: 'first movie of the year',
		overview:
			'overview of the video for the m ',		
		genres: ['Action', 'Horror'],
		budget: 1000,
		poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
		release_date: '2023-01-01',
		revenue: 136907000,
		runtime: 134,
		tagline: "new and best",
		
		vote_average: 9.9,
		vote_count: 1145,
	},

	totalAmount: 1,
};
