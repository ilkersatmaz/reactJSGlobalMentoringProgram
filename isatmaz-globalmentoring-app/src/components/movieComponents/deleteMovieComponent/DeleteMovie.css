.delete-form{
    padding: 60px;
    background-color: #232323;
}
.delete-form-header{
    font-size: 40px;
    color:#FFFFFF;
    margin: 0;
    margin-bottom: 40px;
}
.delete-form-div{
    display: grid;
    grid-template-columns: auto;
    column-gap: 50px;
    row-gap: 30px;
}

.confirm-button{
    padding: 10px 20px;
    border: 1px solid transparent;
    background-color: #f65261;
    color: #FFFFFF;    
    justify-self: end;
    width: 150px;
    text-align: center;
}

.confirmation-text{
    color: #FFFFFF;
    font-size: 20px;
}