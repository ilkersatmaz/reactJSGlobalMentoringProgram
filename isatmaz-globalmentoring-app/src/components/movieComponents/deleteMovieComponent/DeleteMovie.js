import React from 'react';
import Popup from 'reactjs-popup';
import './DeleteMovie.css';
import ConfirmationPopup from '../confirmationComponent/ConfirmationPopup';

function DeleteMovie () {

  const handleSubmit = (event) => {
    alert('movie was removed from the list: ' + this.state.value);
    event.preventDefault();
  }
  return (
    <form className='delete-form' onSubmit={()=>{handleSubmit()}}>
      <p className='delete-form-header'> DELETE MOVIE</p>
      <div className='delete-form-div'>
        <p className='confirmation-text'>
          Are you sure you want to delete this movie?
        </p>
        <Popup className="confirmation-popup" trigger={<button className='confirm-button'>CONFIRM</button>} position="bottom left" arrow={false}>
            <ConfirmationPopup />
        </Popup>                  
      </div>        
      
    </form>
  );
}

export default DeleteMovie