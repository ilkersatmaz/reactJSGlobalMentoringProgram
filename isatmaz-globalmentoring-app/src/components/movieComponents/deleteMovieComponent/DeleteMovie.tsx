import React from 'react';
import { useAppDispatch } from '../../../hooks';
import { deleteSelectedMovie } from '../../../store/movies/movieOperationSlice';
import './DeleteMovie.css';

function DeleteMovie (props:any) {
  const {setIsConfirmationOpen, movieId} = props;
  const dispatch = useAppDispatch();
  
  
  const handleSubmit = (event: any) => {
    dispatch(deleteSelectedMovie(movieId));
    setIsConfirmationOpen(false);
  }
  return (
    <form className='delete-form' onSubmit={handleSubmit}>
      <p className='delete-form-header'> DELETE MOVIE</p>
      <div className='delete-form-div'>
        <p className='confirmation-text'>
          Are you sure you want to delete this movie?
        </p>
        <button className='confirm-button'>CONFIRM</button>               
      </div>        

    </form>
  );
}

export default DeleteMovie