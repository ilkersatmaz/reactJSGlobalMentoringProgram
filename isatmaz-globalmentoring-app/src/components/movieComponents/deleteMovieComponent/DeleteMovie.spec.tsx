import { render } from '@testing-library/react';
import React from 'react';
import DeleteMovie from './DeleteMovie';
import { movieResponse } from '../../../../__mocks__/response-mock';

jest.mock('../../../hooks',() => ({
  useAppSelector: () => jest.fn(),
  useAppDispatch: () => jest.fn()
}))

describe('DeleteMovie component', () => {
  it("should match the snapshot", () => {  
    const setIsConfirmationOpen=jest.fn();     
    const { asFragment } = render(<DeleteMovie movieId={movieResponse.data.id} setIsConfirmationOpen={setIsConfirmationOpen}/>);    
    expect(asFragment).toMatchSnapshot();
  });
  it("should work for null", () => {    
    const { asFragment } = render(<DeleteMovie movieId={movieResponse.data.id}/>);    
    expect(asFragment).toMatchSnapshot();
  });
})