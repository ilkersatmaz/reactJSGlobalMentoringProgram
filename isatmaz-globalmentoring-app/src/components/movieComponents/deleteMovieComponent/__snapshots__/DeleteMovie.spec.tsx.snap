// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovie component should match the snapshot 1`] = `[Function]`;

exports[`DeleteMovie component should work for null 1`] = `[Function]`;
